test/integration/taskBasedStatusUpdate.test.js

```diff
@@ -287,67 +287,117 @@ describe("Task Based Status Updates", function () {
-      expect(response.data.usersOnboardingOrAlreadyIdle).to.equal(2);
+      const response = res.body.data;
+      expect(response).to.have.all.keys(
+        "totalUsers",
+        "totalUnprocessedUsers",
+        "totalOnboardingUsersAltered",
+        "totalOnboardingUsersUnAltered",
+        "totalActiveUsersAltered",
+        "totalActiveUsersUnAltered",
+        "totalIdleUsersAltered",
+        "totalIdleUsersUnAltered"
+      );
+      expect(response.totalUsers).to.equal(10);
+      expect(response.totalUnprocessedUsers).to.equal(0);
+      expect(response.totalOnboardingUsersAltered).to.equal(1);
+      expect(response.totalOnboardingUsersUnAltered).to.equal(1);
+      expect(response.totalActiveUsersAltered).to.equal(3);
+      expect(response.totalActiveUsersUnAltered).to.equal(1);
+      expect(response.totalIdleUsersAltered).to.equal(3);
+      expect(response.totalIdleUsersUnAltered).to.equal(1);
+
+      const userStatus000Data = (await userStatusModel.doc("userStatus000").get()).data();
+      expect(userStatus000Data.currentStatus.state).to.equal(userState.IDLE);
       const userStatus001Data = (await userStatusModel.doc("userStatus001").get()).data();
-      expect(userStatus001Data.currentStatus.state).to.equal(userState.IDLE);
+      expect(userStatus001Data.currentStatus.state).to.equal(userState.OOO);
+      expect(userStatus001Data.futureStatus.state).to.equal(userState.IDLE);
       const userStatus002Data = (await userStatusModel.doc("userStatus002").get()).data();
-      expect(userStatus002Data.currentStatus.state).to.equal(userState.OOO);
-      expect(userStatus002Data.futureStatus.state).to.equal(userState.IDLE);
+      expect(userStatus002Data.currentStatus.state).to.equal(userState.IDLE);
       const userStatus003Data = (await userStatusModel.doc("userStatus003").get()).data();
-      expect(userStatus003Data.currentStatus.state).to.equal(userState.IDLE);
-      const userStatus004Data = (await userStatusModel.doc("userStatus004").get()).data();
-      expect(userStatus004Data.currentStatus.state).to.equal(userState.ONBOARDING);
-      const userStatus005SnapShot = await userStatusModel.where("userId", "==", userId5).limit(1).get();
-      const [userStatus005Doc] = userStatus005SnapShot.docs;
-      const userStatus005Data = userStatus005Doc.data();
-      expect(userStatus005Data.currentStatus.state).to.equal(userState.IDLE);
+      expect(userStatus003Data.currentStatus.state).to.equal(userState.ONBOARDING);
+      const userStatus004SnapShot = await userStatusModel.where("userId", "==", userId4).limit(1).get();
+      const [userStatus004Doc] = userStatus004SnapShot.docs;
+      const userStatus004Data = userStatus004Doc.data();
+      expect(userStatus004Data.currentStatus.state).to.equal(userState.IDLE);
+      const userStatus005Data = (await userStatusModel.doc("userStatus005").get()).data();
+      expect(userStatus005Data.currentStatus.state).to.equal(userState.ACTIVE);
```

Comment:
Nit: Why it is active not idle?? I think May be you are going to change something Batch update code .